File: Leetcode - Java/145_tree_postorderTraversal.java
// https://leetcode.com/problems/binary-tree-postorder-traversal/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        ArrayList<Integer> postorder = new ArrayList<Integer>();
        if(root != null) {
            postorder.addAll(postorderTraversal(root.left));
            postorder.addAll(postorderTraversal(root.right));
            postorder.add(root.val);
        }
        return postorder; 
    }
}

File: Leetcode-Python-New/145_tree_postorderTraversal.py
# https://leetcode.com/problems/binary-tree-postorder-traversal/

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        def postorder(node: TreeNode) -> None:
            if node is None:
                return
            postorder(node.left)
            postorder(node.right)
            postorderList.append(node.val)
        postorderList = []
        postorder(root)
        return postorderList